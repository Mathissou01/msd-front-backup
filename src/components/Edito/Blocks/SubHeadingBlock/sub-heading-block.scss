@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.wysiwyg" as wysiwyg;

.c-SubHeadingBlock {
  @include wysiwyg.html-parse;

  display: flex;
  flex-direction: column;
  width: 100%;
}
