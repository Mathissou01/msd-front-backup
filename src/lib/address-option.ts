export interface AddressOption {
  id: string;
  city: string;
  citycode?: string;
  context?: string;
  importance?: number;
  label?: string;
  name?: string;
  postcode: string;
  score?: number;
  type?: string;
  x?: number;
  y?: number;
  houseNumber?: number;
  street?: string;
  __typename?: string;
}
