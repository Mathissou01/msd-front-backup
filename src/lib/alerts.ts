interface ICoordinates {
  longitude: number;
  latitude: number;
}

interface IAddressInfo {
  city: string;
  street: string;
}
export type { ICoordinates, IAddressInfo };
