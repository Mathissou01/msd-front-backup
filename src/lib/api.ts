import qs from "qs";
import Router from "next/router";

export function getStrapiURL(path = "") {
  return `${process.env.NEXT_PUBLIC_API_URL}${path}`;
}

export async function fetchAPI(
  path: string,
  urlParamsObject = {},
  options = {},
) {
  const mergedOptions = {
    headers: {
      "Content-Type": "application/json",
    },
    ...options,
  };

  const queryString = qs.stringify(urlParamsObject);
  const requestUrl = `${getStrapiURL(
    `/api${path}${queryString ? `?${queryString}` : ""}`,
  )}`;

  const response = await fetch(requestUrl, mergedOptions);

  if (!response.ok) {
    console.error(response.statusText);
    throw new Error(`An error occurred please try again`);
  }
  return await response.json();
}

interface IFetchGraphQLParams {
  query: {};
  baseUrl?: string;
  variables?: {};
  urlParamsObject?: {};
  options?: {};
}

export async function fetchGraphQL(params: IFetchGraphQLParams) {
  const mergedOptions = {
    method: "POST",
    headers: {
      Authorization: `bearer ${process.env.STRAPI_API_TOKEN}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      query: params.query,
      variables: params.variables,
    }),
    ...params.options,
  };

  const queryString = qs.stringify(params.urlParamsObject);
  const requestUrl = getStrapiURL(
    `/graphql${queryString ? `?${queryString}` : ""}`,
  );

  return await fetch(requestUrl, mergedOptions).then((response) => {
    if (!response.ok) {
      throw new Error(`An error occurred please try again`);
    }
    return response.json();
  });
}

export async function fetchGraphQLDynamic(params: IFetchGraphQLParams) {
  const mergedOptions = {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      query: params.query,
      variables: params.variables,
    }),
    ...params.options,
  };

  const queryString = qs.stringify(params.urlParamsObject);
  const requestUrl = `${
    params.baseUrl ?? process.env.NEXT_PUBLIC_BASE_URL ?? ""
  }/api/graphql${queryString ? `?${queryString}` : ""}`;

  console.log(queryString);
  console.log(mergedOptions);
  const response = await fetch(requestUrl, mergedOptions);

  if (!response.ok) {
    console.error(response.statusText);
    throw new Error(`An error occurred please try again`);
  }
  return await response.json();
}

export async function getPages(preview: boolean | null = false) {
  const pagesData = await fetchAPI(
    `/pages${preview ? "?publicationState=preview" : ""}`,
  );

  if (pagesData === null || pagesData.data?.length === 0) {
    redirectToHome();
  }

  return pagesData.data;
}

export async function getPageBySlug(
  slug: string,
  preview: boolean | null = false,
) {
  if (!slug) {
    redirectToHome();
  }

  const pageData = await fetchAPI(
    `/pages?filters[slug][$eq]=${slug}${
      preview ? "&publicationState=preview" : ""
    }`,
  );

  if (pageData === null || pageData.data?.length === 0) {
    redirectToHome();
  }

  return pageData.data;
}

export function redirectToHome() {
  if (typeof window !== "undefined") {
    Router.replace("/", "/");
  }
}
