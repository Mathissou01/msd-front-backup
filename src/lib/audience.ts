import { Enum_Audience_Type } from "../graphql/codegen/generated-types";

export interface IAudience {
  id: string;
  type?: Enum_Audience_Type;
}
