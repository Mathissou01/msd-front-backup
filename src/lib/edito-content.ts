import {
  EventEntity,
  FreeContentEntity,
  NewEntity,
  QuizEntity,
  TipEntity,
} from "../graphql/codegen/generated-types";
import { removeNulls } from "./utilities";

export type TEditoFields = "news" | "event" | "freeContent" | "quiz" | "tip";

export const editoFields: Array<TEditoFields> = [
  "news",
  "event",
  "freeContent",
  "quiz",
  "tip",
];

export type TEditoTypenames =
  | "EventEntity"
  | "FreeContentEntity"
  | "NewEntity"
  | "QuizEntity"
  | "TipEntity";

export type TEditoTypes =
  | EventEntity
  | FreeContentEntity
  | NewEntity
  | QuizEntity
  | TipEntity;

export function isEditoType<T extends TEditoTypes>(
  content: TEditoTypes,
  typename: TEditoTypenames,
): content is T {
  return (content as T).__typename === typename;
}

/* Blocks */
export type TBlocksDynamicZone =
  | "ComponentBlocksFile"
  | "ComponentBlocksHorizontalRule"
  | "ComponentBlocksImage"
  | "ComponentBlocksSubHeading"
  | "ComponentBlocksVideo"
  | "ComponentBlocksWysiwyg"
  | "Error";
export type TDynamicFieldOption = Exclude<TBlocksDynamicZone, "Error">;

interface IPartialBlockDynamicZone {
  __typename: TBlocksDynamicZone;
  id: string;
}

interface IPartialBlock {
  __typename: TDynamicFieldOption;
  id: string;
}

export type IEditoBlock =
  | IPartialBlock
  | IBlocksFile
  | IBlocksHorizontalRule
  | IBlocksImage
  | IBlocksSubHeading
  | IBlocksVideo
  | IBlocksWysiwyg;

export interface IBlocksFile extends IPartialBlock {
  __typename: "ComponentBlocksFile";
}

export interface IBlocksHorizontalRule extends IPartialBlock {
  __typename: "ComponentBlocksHorizontalRule";
  hr: string;
}

export interface IBlocksImage extends IPartialBlock {
  __typename: "ComponentBlocksImage";
}

export interface IBlocksSubHeading extends IPartialBlock {
  __typename: "ComponentBlocksSubHeading";
  subHeadingTag?: string;
  subHeadingText?: string;
}

export interface IBlocksVideo extends IPartialBlock {
  __typename: "ComponentBlocksVideo";
  videoLink?: string;
  transcriptText?: string;
}

export interface IBlocksWysiwyg extends IPartialBlock {
  __typename: "ComponentBlocksWysiwyg";
}

/* Methods */
export function isEditoBlock(
  block: Partial<IPartialBlockDynamicZone>,
): block is IEditoBlock {
  return "__typename" in block && "id" in block;
}

export function isEditoBlockTypename<T extends IPartialBlock>(
  block: IEditoBlock,
  typename: TDynamicFieldOption,
): block is T {
  return block.__typename === typename;
}

export function remapEditoBlocksDynamicZone(
  blocks?: Array<Partial<IPartialBlockDynamicZone> | null> | null,
): Array<IEditoBlock> {
  return (
    blocks
      ?.map((block) => {
        if (block) {
          const type = block.__typename;
          if (type && type !== "Error" && isEditoBlock(block)) {
            return {
              ...block,
              __typename: block.__typename,
              id: block.id,
            };
          }
        }
      })
      .filter(removeNulls) ?? []
  );
}
