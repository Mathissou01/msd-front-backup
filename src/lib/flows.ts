export const renderFlowName: (name: string) => string = (name: string) => {
  switch (name) {
    case "OMR":
      return "Ordures ménagères";
    case "CS":
      return "Emballages";
    case "all":
      return "Tous";
    default:
      return name;
  }
};
