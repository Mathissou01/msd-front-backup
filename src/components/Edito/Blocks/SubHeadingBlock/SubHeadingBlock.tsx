import React from "react";
import "./sub-heading-block.scss";

interface ISubHeadingBlockProps {
  subHeadingText: string;
  subHeadingTag?: string;
}

function generateTitleJSX(subHeadingTag: string, subHeadingText: string) {
  switch (subHeadingTag) {
    case "h2":
      return <h2>{subHeadingText}</h2>;
    case "h3":
      return <h3>{subHeadingText}</h3>;
    case "h4":
      return <h4>{subHeadingText}</h4>;
    case "h5":
      return <h5>{subHeadingText}</h5>;
    case "h6":
      return <h6>{subHeadingText}</h6>;
    default:
  }
}

export default function SubHeadingBlock({
  subHeadingText,
  subHeadingTag,
}: ISubHeadingBlockProps) {
  return (
    <div className="c-SubHeadingBlock">
      {subHeadingTag && generateTitleJSX(subHeadingTag, subHeadingText)}
      <div className="c-SubHeadingBlock__Line" />
    </div>
  );
}
