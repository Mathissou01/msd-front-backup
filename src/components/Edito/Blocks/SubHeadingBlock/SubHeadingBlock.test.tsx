import { render } from "@testing-library/react";
import SubHeadingBlock from "./SubHeadingBlock";
import { THeadingTags } from "../../../../lib/edito-content";

const mock = {
  subHeadingTag: "h2" as THeadingTags,
  subHeadingText: "Heading Block",
};
describe("SubHeadingBlock", () => {
  it("renders", () => {
    const { container } = render(
      <SubHeadingBlock
        subHeadingText={mock.subHeadingText}
        subHeadingTag={mock.subHeadingTag}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
