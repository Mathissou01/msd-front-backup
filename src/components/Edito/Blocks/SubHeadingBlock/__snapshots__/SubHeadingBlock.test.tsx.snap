// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeadingBlock renders 1`] = `
<div>
  <div
    class="c-SubHeadingBlock"
  >
    <h2>
      Heading Block
    </h2>
  </div>
</div>
`;
